### PR TITLE
Implement the Media section on the product create/update page

### PR DESCRIPTION
**Pull request by jakubtobiasz**
| Q               | A                                                            |
|-----------------|--------------------------------------------------------------|
| Branch?         | `bootstrap-admin-panel`

Based on #16075.

### CODE REVIEW

File: features/admin/product/managing_products/adding_images_to_existing_product.feature [hunk 1]
```diff
@@ -75,7 +75,7 @@ Feature: Adding images to an existing product
         Then I should be notified that it has been successfully uploaded
         And this product should have only one image
 
-    @todo @ui @mink:chromedriver @api
+    @ui @mink:chromedriver
```
Comment (Wojdylak) [inline features/admin/product/managing_products/adding_images_to_existing_product.feature]: Why did you remove `@api`?

File: src/Sylius/Bundle/CoreBundle/Form/Type/Product/ProductImageType.php [hunk 2]
```diff
@@ -13,29 +13,43 @@
...
 use Sylius\Component\Core\Model\ProductInterface;
+use Symfony\Bridge\Doctrine\Form\Type\EntityType;
 use Symfony\Component\Form\FormBuilderInterface;
 use Symfony\Component\Form\FormInterface;
 use Symfony\Component\Form\FormView;
 use Symfony\Component\OptionsResolver\OptionsResolver;
 
 final class ProductImageType extends ImageType
 {
+    public function __construct(string $dataClass, private string $productVariantClass, array $validationGroups = [])
```
Comment (Wojdylak) [inline src/Sylius/Bundle/CoreBundle/Form/Type/Product/ProductImageType.php:28]:
```suggestion
    public function __construct(
    string $dataClass, 
    private string $productVariantClass, 
    array $validationGroups = [],
    ) {
```

File: src/Sylius/Bundle/AdminBundle/Resources/views/Product/Tab/_media.html.twig
Comment (Wojdylak) [inline src/Sylius/Bundle/AdminBundle/Resources/views/Product/Tab/_media.html.twig:1]: Empty?

File: src/Sylius/Behat/Service/Helper/AutocompleteHelper.php [hunk 3]
```diff
@@ -55,16 +56,17 @@ public function search(DriverInterface $driver, string $selector, string $search
...
-                element.parentElement.querySelectorAll('[data-selectable]').forEach((node) => searchResults[node.textContent] = node.dataset.value);
+                element.parentElement.querySelectorAll('[data-selectable]').forEach((node) => searchResults[node.dataset.value] = node.textContent);
 
                 return searchResults;
             })();
         SCRIPT);
     }
 
-    public function select(DriverInterface $driver, string $selector, string $value): void
+    public function selectByName(DriverInterface $driver, string $selector, string $value): void
```
Comment (Wojdylak) [inline src/Sylius/Behat/Service/Helper/AutocompleteHelper.php]:
```suggestion
    public function selectByName(DriverInterface $driver, string $selector, string $name): void
```

File: src/Sylius/Bundle/CoreBundle/Form/Type/Product/ProductImageType.php (hunk 2 above)
Comment (jakubtobiasz) [reply src/Sylius/Bundle/CoreBundle/Form/Type/Product/ProductImageType.php:28]: It's not too long yet.

File: src/Sylius/Bundle/AdminBundle/Resources/views/Product/Tab/_media.html.twig
Comment (jakubtobiasz) [reply src/Sylius/Bundle/AdminBundle/Resources/views/Product/Tab/_media.html.twig:1]: Fixed

File: src/Sylius/Behat/Service/Helper/AutocompleteHelper.php (hunk 3 above)
Comment (jakubtobiasz) [reply src/Sylius/Behat/Service/Helper/AutocompleteHelper.php]: Fixed

File: features/admin/product/managing_products/adding_images_to_existing_product.feature (hunk 1 above)
Comment (jakubtobiasz) [reply features/admin/product/managing_products/adding_images_to_existing_product.feature]: Fixed

File: src/Sylius/Behat/Page/Admin/Product/FormTrait.php [hunk 4]
```diff
@@ -13,8 +13,11 @@
 
 namespace Sylius\Behat\Page\Admin\Product;
 
+use Behat\Mink\Element\NodeElement;
+use Behat\Mink\Exception\ElementNotFoundException;
 use Sylius\Behat\Service\DriverHelper;
 use Sylius\Component\Core\Model\ChannelInterface;
+use Sylius\Component\Core\Model\ProductVariantInterface;
 
 trait FormTrait
```
Comment (GSadee) [inline src/Sylius/Behat/Page/Admin/Product/FormTrait.php:22]: I would be for replacing a FormTrait by a FormElement 🖖🏻

File: src/Sylius/Behat/Page/Admin/Product/FormTrait.php [hunk 5]
```diff
@@ -206,13 +213,129 @@ public function getNumberOfAttributes(): int
...
             $driver,
             $this->getElement('product_attribute_input')->getXpath(),
             $attributeName,
         );
     }
 
+    /*
+     * Media management
+     */
+
```
Comment (GSadee) [inline src/Sylius/Behat/Page/Admin/Product/FormTrait.php:226]:
```suggestion
```

File: src/Sylius/Bundle/CoreBundle/Form/Extension/ProductTypeExtension.php [hunk 6]
```diff
@@ -53,7 +53,7 @@ public function buildForm(FormBuilderInterface $builder, array $options): void
                 'choices' => array_flip(Product::getVariantSelectionMethodLabels()),
                 'label' => 'sylius.form.product.variant_selection_method',
             ])
-            ->add('images', CollectionType::class, [
+            ->add('images', LiveCollectionType::class, [
```
Comment (GSadee) [inline src/Sylius/Bundle/CoreBundle/Form/Extension/ProductTypeExtension.php]: It shouldn't be changed in Core to LiveCollection

File: src/Sylius/Behat/Page/Admin/Product/FormTrait.php (hunk 4 above)
Comment (jakubtobiasz) [reply src/Sylius/Behat/Page/Admin/Product/FormTrait.php:22]: I'm planning it for our global unification.

File: src/Sylius/Behat/Page/Admin/Product/FormTrait.php (hunk 5 above)
Comment (jakubtobiasz) [reply src/Sylius/Behat/Page/Admin/Product/FormTrait.php:226]: This class will disappear once unificated (form element), so let's skip it.

File: src/Sylius/Bundle/CoreBundle/Form/Extension/ProductTypeExtension.php (hunk 6 above)
Comment (jakubtobiasz) [reply src/Sylius/Bundle/CoreBundle/Form/Extension/ProductTypeExtension.php]: Done

**Review by GSadee — APPROVED** [review]

File: src/Sylius/Bundle/AdminBundle/Form/Extension/ProductImageTypeExtension.php [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+                    'label' => 'sylius.ui.product_variants',
+                    'multiple' => true,
+                    'required' => false,
+                    'choice_label' => 'descriptor',
+                    'choice_value' => 'code',
+                    'query_builder' => function (EntityRepository $er) use ($options): QueryBuilder {
+                        return $er->createQueryBuilder('o')
+                            ->where('o.product = :product')
+                            ->setParameter('product', $options['product'])
+                            ;
```
Comment (GSadee) [inline src/Sylius/Bundle/AdminBundle/Form/Extension/ProductImageTypeExtension.php:45]:
```suggestion
                        ;
```

**Review by GSadee — APPROVED** [review]
